pennylane/hf/convert.py

```diff
@@ -0,0 +1,316 @@
+    r"""Converts a 2-tuple of complex coefficients and PennyLane operations to
+    OpenFermion ``QubitOperator``.
+
+    This function is the inverse of ``_qubit_operator_to_terms``.
+
+    Args:
+        coeffs (array[complex]):
+            coefficients for each observable, same length as ops
+        ops (Iterable[pennylane.operation.Observable]): List of PennyLane observables as
+            Tensor products of Pauli observables
+        wires (Wires, list, tuple, dict): Custom wire mapping used to convert to qubit operator
+            from an observable terms measurable in a PennyLane ansatz.
+            For types Wires/list/tuple, each item in the iterable represents a wire label
+            corresponding to the qubit number equal to its index.
+            For type dict, only consecutive-int-valued dict (for wire-to-qubit conversion) is
+            accepted. If None, will map sorted wires from all `ops` to consecutive int.
+
+    Returns:
+        QubitOperator: an instance of OpenFermion's ``QubitOperator``.
+
+    **Example**
+
+    >>> coeffs = np.array([0.1, 0.2])
+    >>> ops = [
+    ...     qml.operation.Tensor(qml.PauliX(wires=['w0'])),
+    ...     qml.operation.Tensor(qml.PauliY(wires=['w0']), qml.PauliZ(wires=['w2']))
+    ... ]
+    >>> _pennylane_to_openfermion(coeffs, ops, wires=Wires(['w0', 'w1', 'w2']))
+    0.1 [X0] +
+    0.2 [Y0 Z2]
+    """
+    all_wires = Wires.all_wires([op.wires for op in ops], sort=True)
+
+    if wires is not None:
+        qubit_indexed_wires = _process_wires(
+            wires,
+        )
+        if not set(all_wires).issubset(set(qubit_indexed_wires)):
+            raise ValueError("Supplied `wires` does not cover all wires defined in `ops`.")
+    else:
+        qubit_indexed_wires = all_wires
+
+    q_op = openfermion.QubitOperator()
+    for coeff, op in zip(coeffs, ops):
+
+        if not isinstance(op, qml.operation.Tensor):
+            op = qml.operation.Tensor(op)
+
+        extra_obsvbs = set(op.name) - {"PauliX", "PauliY", "PauliZ", "Identity"}
+        if extra_obsvbs != set():
+            raise ValueError(
+                f"Expected only PennyLane observables PauliX/Y/Z or Identity, "
+                f"but also got {extra_obsvbs}."
+            )
+
+        # Pauli axis names, note s[-1] expects only 'Pauli{X,Y,Z}'
+        pauli_names = [s[-1] if s != "Identity" else s for s in op.name]
+
+        all_identity = all(obs.name == "Identity" for obs in op.obs)
+        if (op.name == ["Identity"] and len(op.wires) == 1) or all_identity:
+            term_str = ""
+        else:
+            term_str = " ".join(
+                [
+                    f"{pauli}{qubit_indexed_wires.index(wire)}"
+                    for pauli, wire in zip(pauli_names, op.wires)
+                    if pauli != "Identity"
+                ]
+            )
+
+        # This is how one makes QubitOperator in OpenFermion
+        q_op += coeff * openfermion.QubitOperator(term_str)
```

Comment:
Can you test this function for a Hamiltonian with a single wire (say tapered H2 one)? I remember getting an error due to a type-casting happening at this line.

Reply:
Added tests of Hamiltonians with a single wire for both `_pennylane_to_openfermion` and `import_observable`.

Reply:
I still get the following error for this Hamiltonian. Adding a complex typecasting in `coeff` as `complex(coeff)` fixes the issue for me. 
![image](https://user-images.githubusercontent.com/23234279/156214450-df1de322-0d6a-4301-973b-dcc92316c597.png)

Reply:
@obliviateandsurrender Can you please copy the code, so I can try it?

Reply:
Sent the notebook on slack.